### PR TITLE
Adding a spring component #112

### PR DESCRIPTION
**Pull request by gftruj**
Cannon.js contains a simple [spring class](http://schteppe.github.io/cannon.js/docs/classes/Spring.html), which could be a nice addition to the a-frame physics system 😄 

I tried to maintain a similar fashion as the 
>> src/components/constraint.js

Basically upon creation i wait for the Cannon bodies to load -> create the spring instance -> update all properties from the component's schema.

If you like it, please let me know:
- which files should i attach to this PR
- what tests do you want for this PR ( except my [simple example](https://jsfiddle.net/3fcynuwv/) with different attributes applied )

### CODE REVIEW

File: src/components/spring.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    damping: {default: 1, min: 0, max: 1},
+
+    // offsets
+    localAnchorA: {type: 'vec3', default: {x: 0, y: 0, z: 0}},
+    localAnchorB: {type: 'vec3', default: {x: 0, y: 0, z: 0}},
+  },
+
+  init: function() {
+    this.system = this.el.sceneEl.systems.physics
+    this.world = this.system.driver.world
```
Comment (donmccurdy) [inline src/components/spring.js]: Is it possible to implement this without touching `system.driver` or `system.driver.world`? If so, it makes it much easier to adapt this to work with Web Workers or future physics engines... in theory you should be able to do `this.system.addComponent(this)` in init, and `this.system.removeComponent(this)` in remove, and then define `beforeStep` and `afterStep` methods on the component where we update the spring. See `system.js`.

File: src/components/spring.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    this.spring = /* {CANNON.Spring} */ null
+  },
+
+  update: function(oldData, newData) {
```
Comment (donmccurdy) [inline src/components/spring.js]: Shouldn't need `newData` here, just `this.data`.

**Comment by donmccurdy** [comment]
Very nice thanks! If you want to include a simple example like that, I think it'd be great. :)

File: src/components/spring.js (hunk 1 above)
Comment (gftruj) [reply src/components/spring.js]: Thanks for the hint, not only did I reimplement it using the system callbacks, but also it seems to be working 😄

**Comment by gftruj** [comment]
Hey, I have no clue why i was sure about those "max" restrictions. I've also utilized the `step()` callback from the physics system. Also i've added a bool to keep the functionality of `play` and `pause`.

**Comment by donmccurdy** [comment]
Nice work, thanks! :)

**Pull request merged**